addons/website/static/src/snippets/s_website_form/options.js

```diff
@@ -1214,16 +1223,32 @@ options.registry.WebsiteFieldEditor = FieldEditor.extend({
                 <we-button data-select-data-attribute="!fileSet">${_t("Is not set")}</we-button>
             </we-select>
         `)[0]);
+        const recursiveFindCircular = (el) => {
+            const dependencyEl = this._getDependencyEl(el);
+            if (!dependencyEl) {
+                return false;
+            }
+            if (dependencyEl.closest('.s_website_form_field').dataset.visibilityDependency === this._getFieldName()) {
+                return true;
+            } else {
+                return recursiveFindCircular(dependencyEl.closest('.s_website_form_field'));
+            }
+        };
 
         // Update available visibility dependencies
         const selectDependencyEl = uiFragment.querySelector('we-select[data-name="hidden_condition_opt"]');
         if (selectDependencyEl) {
             const existingDependencyNames = [];
-            for (const el of this.formEl.querySelectorAll('.s_website_form_field:not(.s_website_form_dnone)')) {
+            const currentFieldName = this._getFieldName();
+            // Exclude field that would create a direct circular dependency
+            const preventCircularSelector = `:not([data-visibility-dependency="${currentFieldName}"])`;
+            for (const el of this.formEl.querySelectorAll(`.s_website_form_field:not(.s_website_form_dnone)${preventCircularSelector}`)) {
```

Comment:
I wonder if it would not be easier to just use the recursive method. The code would be smaller (so easier to read) and I wonder if it would not actually be faster. Parsing and considering your `:not` is probably the most costly operation here (but the code being smaller and easier to read is the real argument).

Reply:
Sure :+1: 
That will also avoid the notion of direct and indirect circular dependency.
I needed the recursive method to be a bit tweaked/logical reversed, done.